### ISSUE TITLE
VirtualRequestLogger should not log sensitive information

### ISSUE THREAD

**Issue by oehmiche** [zepgram/module-rest#7]
Hello Zepgram-Team,

with activated debug-logging we just noticed, that the HTTP Request-Response logging (which is produced by the HttpClient) does also contain authorization information provided using the HTTP header. 
In these times, sensitive information like access credentials should not be written to logfiles.

We solved this by overwriting the `Zepgram\Rest\Logger\VirtualRestLogger` and adding a Monolog `Processor` which sanitizes the logging context:

```
   <virtualType name="Zepgram\Rest\Logger\VirtualRestLogger" type="Magento\Framework\Logger\Monolog">
        <arguments>
            <argument name="handlers" xsi:type="array">
                <item name="system" xsi:type="object">Zepgram\Rest\Logger\Handler</item>
            </argument>
            <argument name="processors" xsi:type="array">
                <item name="http-header-sanitize" xsi:type="object">SomeVendor\ReduceZepgramLogging\Logger\SanitizeHttpHeaderPreprocessor</item>
            </argument>
        </arguments>
    </virtualType>
```

```
<?php
declare(strict_types=1);

namespace SomeVendor\ReduceZepgramLogging\Logger;

use Monolog\Processor\ProcessorInterface;

class SanitizeHttpHeaderPreprocessor implements ProcessorInterface
{

    private const AUTHORIZATION_HEADER = 'Authorization';
    private const UNSAFE_HEADERS = [self::AUTHORIZATION_HEADER];

    /**
     * @inheritDoc
     */
    public function __invoke(array $record): array
    {
        if (isset($record['context']['request']['options']['headers'])) {
            foreach (self::UNSAFE_HEADERS as $unsafeHeader) {
                $record['context']['request']['options']['headers'][$unsafeHeader] = '***';
            }
        }
        return $record;
    }
}
```

Perhaps it's worth to integrate this into the `module-rest` code-base. - What do you think?

Kind regards,
Odilo

**Comment by zepgram** [comment]
Hello,

Yes you are right. Historically, I'm doing this for my customers in a dedicated generic module which obfuscate all sensitive data.

But I should definitely handle it natively with this module.

Thanks for reporting this.

**Comment by zepgram** [comment]
Solved with #8

**Comment by oehmiche** [comment]
Super nice. Thanks for providing the fix so quickly.
